src/platform/forms-system/src/js/review/submit-states/ValidationError.jsx

```diff
@@ -1,24 +1,142 @@
+import { focusAndScrollToReviewElement } from '../../utilities/ui';
+import { openReviewChapter, setEditMode } from '../../actions';
+
+function ValidationError(props) {
+  const {
+    appType,
+    buttonText,
+    formConfig,
+    onBack,
+    onSubmit,
+    testId,
+    form,
+  } = props;
+
+  const errorRef = useRef(null);
+  const errors = form.formErrors?.errors || [];
+  const errorsLen = errors.length;
+  const [hadErrors, setHadErrors] = useState(false);
+
+  // error links need evaluation & testing before production
+  const renderErrors =
+    (errorsLen > 0 && formConfig.showReviewErrors?.()) || false;
+
+  useEffect(
+    () => {
+      // Move focus to legend
+      if (renderErrors && !errorRef.current.classList.contains('has-focused')) {
+        // initially focus on alert legend immediately above error links
+        errorRef.current.focus();
+        errorRef.current.classList.add('has-focused');
+      }
+    },
+    [renderErrors, errorRef],
+  );
+
+  let alert;
+  if (renderErrors) {
+    if (!hadErrors && errorsLen > 0) {
+      setHadErrors(true);
+    }
+
+    const resolved = hadErrors && errorsLen === 0;
+
+    // Once all errors have been resolved, we can update the message
+    const errorMessage = resolved ? (
+      `Try submitting your ${appType} again.`
+    ) : (
+      <>
+        <p aria-describedby="missing-info-alert-legend">
+          You’ll need to fill in the missing information before you can submit
+          your {appType}
+        </p>
+        <fieldset>
+          <legend
+            id="missing-info-alert-legend"
+            className="vads-u-font-size--base"
+          >
+            {`Please return to the following ${
+              errorsLen === 1 ? 'part' : `${errorsLen} parts`
+            } of the form:`}
+          </legend>
+          <ul className="vads-u-margin-left--2 error-message-list">
+            {errors.map(error => (
+              <li key={error.name}>
+                {error.chapterKey ? (
+                  <a
+                    href="#"
+                    onClick={event => {
+                      event.preventDefault();
+                      props.openReviewChapter(error.chapterKey);
+                      props.setEditMode(
+                        error.pageKey,
+                        true, // enable edit mode
+                        error.index || null,
+                      );
+                      // props.formContext.onError();
+                      focusAndScrollToReviewElement(error);
+                    }}
+                  >
+                    {error.message}
+                  </a>
+                ) : (
+                  error.message
+                )}
+              </li>
+            ))}
+          </ul>
+        </fieldset>
+      </>
+    );
+    alert = (
+      <div
+        className="usa-alert usa-alert-error schemaform-failure-alert"
+        data-testid={testId}
+      >
```

Comment:
I couldn't use the `AlertBox` component here because it doesn't allow the addition of `aria-desribedby` to the header, and I needed to focus on the header after rendering - per Trevor's feedback

Reply:
To me this sounds like an opportunity/reason to improve our alert box component,

Reply:
@1Copenut can you remember if you left this feedback on an issue anywhere? I'd like to address it in an in-progress PR.

Reply:
Agreed. If a component in the design system needs an accessibility fix like this, we should make the update to the design system instead of going around it (more or less).

That said, is this still a concern with the [`<va-alert>` web component](https://design.va.gov/storybook/?path=/docs/components-va-alert--default), @bkjohnson?

Reply:
It is still a concern - I couldn't find any more detail on the problem